src/modules/explorer/components/ProposalSelectionMenu.tsx

```diff
@@ -78,12 +75,9 @@ enum ProposalModalKey {
 //   ],
 // };
 
-export const ProposalSelectionMenu: React.FC<Props> = ({
-                                                         open,
-                                                         handleClose,
-                                                       }) => {
+export const ProposalSelectionMenu: React.FC<Props> = ({ open, handleClose, daoType }) => {
   const daoId = useDAOID();
-  const {data: dao} = useDAO(daoId);
+  const { data: dao } = useDAO(daoId);
   const [openModal, setOpenModal] = useState<ProposalModalKey>();
```

Comment:
We already have the DAO type from the `useDAO` hook, no need to pass `daoType`.

`BaseDAO` class is abstract, it cannot be instantiated, only inherited from. Therefore there is no case where a DAO object is not an instance of either a `RegistryDAO` or a `TreasuryDAO`. To check this, you can use `instanceof` Typescript operator or make a static method on the `BaseDAO` class that checks for this.